rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClient.java

```diff
@@ -145,4 +146,7 @@
 
     VersionSearchResults searchVersions(String artifactId, Integer offset, Integer limit);
 
+    void setNextRequestHeaders(Map<String, String> headers);
+
+    Map<String, String> getHeaders();
```

Comment:
I didn't expect to see this based on our design session on this feature.  I expected `setNextRequestHeaders` but not `getHeaders`.  I may be forgetting something... what is the use-case for this?  My recollection is that `getHeaders` was my (rejected) suggestion for allowing per-request headers.  But I thought we were all in favor of Fabian's idea.

Reply:
This is Fabian's suggestion with that minor addition of the `getHeaders` method. The use case I see is to compare the existing headers with the map provided by the user. I just thought that it could be a good addition. I can remove it if you don't agree.

Reply:
Who would be doing that comparison?  At what point in the flow/usage of the client?

Reply:
If you're reusing threads maybe you want to check the already existing headers before setting new ones. As I said, just thought it could be a good addition, I don't have a strong opinion about this.

Reply:
@Apicurio/developers Any thoughts on this?

Reply:
Now that I understand the purpose, I also don't have a strong opinion on it.  :)

Reply:
in response to "If you're reusing threads " , if I'm not mistaken, the headers are wiped out after every request, so even re-using threads , always, after a request `getHeaders` will return an empty map, no?
I thought you added the `getHeaders` for testing purposes, which I can partly understand ( there are other workarounds for testing the headers)

Reply:
anyway, I'm ok with this :)

Reply:
I realized that yesterday after the discussion with Eric. Yes, you're right, even if re-using threads, the headers are wiped out after every request. That leads us to the only use case of the test purpose.

Reply:
for now, we could add some documentation warning that `getHeaders` method is intended to only be used for testing, and we can work later on removing that method and implement a different testing approach